features/admin.applications.v1/hooks/use-application-api-authorization.ts

```diff
@@ -0,0 +1,125 @@
+import { useEffect, useState } from "react";
+import useSubscribedAPIResources from "../api/use-subscribed-api-resources";
+import { ApplicationManagementConstants } from "../constants/application-management";
+import {
+    AuthorizedAPIListItemInterface,
+    AuthorizedPermissionListItemInterface
+} from "../models/api-authorization";
+
+interface UseAPIAuthorizationReturn {
+    allAuthorizedScopes: AuthorizedPermissionListItemInterface[];
+    subscribedAPIResourcesListData: AuthorizedAPIListItemInterface[];
+    isSubscribedAPIResourcesListLoading: boolean;
+    subscribedAPIResourcesFetchRequestError: any;
+    mutateSubscribedAPIResourcesList: () => void;
+    bulkChangeAllAuthorizedScopes: (updatedScopes: AuthorizedPermissionListItemInterface[], removed: boolean) => void;
+}
+
+/**
+ * Hook to manage API authorization logic.
+ *
+ * @param appId - Application ID
+ * @param updateApiScopes - Function to update API scopes in context
+ * @returns API authorization state and functions
+ */
+export const useAPIAuthorization = (
+    appId: string,
+    updateApiScopes?: (scopes: string) => void
+): UseAPIAuthorizationReturn => {
+    const [ allAuthorizedScopes, setAllAuthorizedScopes ] = useState<AuthorizedPermissionListItemInterface[]>([]);
+    const { isSubOrganization } = useGetCurrentOrganizationType();
+    const requiredApplicationViewScope: string = isSubOrganization
+        ? ApplicationManagementConstants.INTERNAL_ORG_APPLICATION_MGT_VIEW
+        : ApplicationManagementConstants.INTERNAL_APPLICATION_MGT_VIEW;
+    const hasApplicationViewPermission: boolean = useRequiredScopes([ requiredApplicationViewScope ]);
+
+    const {
+        data: subscribedAPIResourcesListData,
+        isLoading: isSubscribedAPIResourcesListLoading,
+        error: subscribedAPIResourcesFetchRequestError,
+        mutate: mutateSubscribedAPIResourcesList
+    } = useSubscribedAPIResources(appId, hasApplicationViewPermission);
+
+    /**
+     * Initalize the all authorized scopes.
+     */
+    useEffect(() => {
+        if (subscribedAPIResourcesListData?.length > 0) {
+            let authorizedScopes: AuthorizedPermissionListItemInterface[] = [];
+
+            subscribedAPIResourcesListData.forEach((subscribedAPIResource: AuthorizedAPIListItemInterface) => {
+                authorizedScopes = authorizedScopes.concat(subscribedAPIResource.authorizedScopes);
+            });
+
+            setAllAuthorizedScopes(authorizedScopes);
+        } else if (subscribedAPIResourcesListData?.length === 0) {
+            setAllAuthorizedScopes([]);
+        }
+    }, [ subscribedAPIResourcesListData ]);
+
+    /**
+     * Update API Scopes in context.
+     */
+    useEffect(() => {
+        if (allAuthorizedScopes) {
+            updateApiScopes(allAuthorizedScopes.map(
+                (scope: AuthorizedPermissionListItemInterface) => scope.name).join(" ")
+            );
+        }
+    }, [ allAuthorizedScopes ]);
```

Comment:
Let's avoid unnecessary states and use `useMemo`.
Ex:
```
const allAuthorizedScopeNames: string = useMemo(
  () => allAuthorizedScopes?.map((s) => s.name)?.join(" ") ?? "",
  [allAuthorizedScopes]
);
```
And directly expose `allAuthorizedScopes`, `allAuthorizedScopeNames`, and please do necessary adjustments.
And no need to use the callback `updateApiScopes`.